src/evict/evict_lru.c

```diff
@@ -284,10 +282,9 @@ __wt_evict_thread_run(WT_SESSION_IMPL *session, WT_THREAD *thread)
      * busy and then opens a different file (in this case, the HS file), it can deadlock with a
      * thread waiting for the first file to drain from the eviction queue. See WT-5946 for details.
      */
-    if (!F_ISSET(conn, WT_CONN_IN_MEMORY)) {
-        session_flags = 0; /* [-Werror=maybe-uninitialized] */
-        WT_RET(__wt_hs_cursor(session, &session_flags, &is_owner));
-        WT_RET(__wt_hs_cursor_close(session, session_flags, is_owner));
+    if (session->hs_cursor == NULL && !F_ISSET(conn, WT_CONN_IN_MEMORY | WT_CONN_READONLY)) {
```

Comment:
Why the addition of WT_CONN_READONLY here?

Reply:
I had noticed elsewhere (e.g., `wt_hs_create()`) that we don't set up or use the history store if either `WT_CONN_IN_MEMORY` or `WT_CONN_READONLY` is set. So I updated the condition here to match.

Reply:
The reason is in-memory connection and read-only do not create a history store.

Reply:
You could still be running in read only and attempt to read something from the history store in a previous iteration no?

Reply:
Ah.  Yes, of course.  I'll remove the WT_CONN_READONLY check here.